src/libs/actions/Session.js

```diff
@@ -3,16 +3,53 @@ import Str from 'expensify-common/lib/str';
 import _ from 'underscore';
 import ONYXKEYS from '../../ONYXKEYS';
 import redirectToSignIn from './SignInRedirect';
+import * as Network from '../Network';
 import * as API from '../API';
 import CONFIG from '../../CONFIG';
 import PushNotification from '../Notification/PushNotification';
 import ROUTES from '../../ROUTES';
 
 let credentials;
-Onyx.connect({
-    key: ONYXKEYS.CREDENTIALS,
-    callback: val => credentials = val,
-});
+
+// Indicates if we're in the process of re-authenticating. When an API call returns jsonCode 407 indicating that the
+// authToken expired, we set this to true, pause all API calls, re-authenticate, and then use the authToken from the
+// response in the subsequent API calls
+let isReauthenticating = false;
+
+// Used to prevent calling CreateLogin more than once since this callback is triggered when we set
+// authToken, loading, error, etc
```

Comment:
> Hm, I see. It feels to me like we're maybe putting too many things inside the `session` Onyx object then

Yup



> For example, in the instance where `INSIDE createLogin true` then shouldn't we just prevent `createLogin()` from being called a second time?

We are with
```js
     if (creatingLogin) {
         return;
     }
```

> What if `session.loading` and `session.error` were moved to a different Onyx key?

We can, do you think that should be part of this PR? I can do it but it'll delay the PR, just like addressing [this comment](https://github.com/Expensify/Expensify.cash/pull/915#discussion_r541080220), both of which aren't part of the original scope, and since I've been working on this refactor for a while now, I'd prefer if we don't keep adding scope to this change. What do you think? I you agree I'll create issue for both of those things

Reply:
> We are with

That's being done from inside of `createLogin()` which is not what I meant. I meant to not call `createLogin()` in the first place if we don't need to.

As it turns out, a lot of this is a non-issue and I'm already having to change a lot of this flow in https://github.com/Expensify/Expensify.cash/pull/1025 (that's the bad part about your PR here still being a WIP). I'm removing `session.loading` but for now I am leaving `session.error`.